File: tuple.js
function solution(s) {
    var ans = [];
    let start = 0;
    let a = new Array(1000000);
    for (let i = 0; i <= 1000000; i++) {
        a[i] = new Array(0);
    }
    let str = "";
    let r = 0;
    let c = 0;
    for (let i = 1; i < s.length - 1; i++) {
        if (s[i] === "{") {
            start = 1;
        } else if (s[i] === "}") {
            a[r][c++] = str;
            str = "";
            start = 0;
            r++;
            c = 0;
        } else if (start === 1 && s[i] === ",") {
            a[r][c++] = str;
            str = "";
        } else if (start === 1) {
            str += s[i];
        }
    }
    a.sort((a, b) => {
        if (a.length === 0 || b.length === 0) {
            return 0;
        }
        return a.length - b.length;
    });
    for (let i = 0; i < a.length; i++) {
        for (let j = 0; j < a[i].length; j++) {
            let flag = 0;
            for (let k = 0; k < ans.length; k++) {
                if (+a[i][j] === ans[k]) {
                    flag = 1;
                    continue;
                }
            }
            if (flag === 0) {
                ans.push(+a[i][j]);
                break;
            }
        }
    }
    return ans;
}

File: tuple.cpp
#include <bits/stdc++.h>
using namespace std;

int check[100001];

bool cmp(vector<int> &a, vector<int> &b) {
    return a.size() < b.size();
}

vector<int> solution(string s) {
    vector<int> answer;
    vector<vector<int>> a;
    for (int i = 1; i < s.size() - 1; i++) {
        if (s[i] == '{') {
            vector<int> tmp;
            string k = "";
            for (int j = i + 1; j < s.size() - 1; j++) {
                if (s[j] == '}') {
                    tmp.push_back(stoi(k));
                    k = "";
                    i = j + 1;
                    break;
                } else if (s[j] == ',') {
                    tmp.push_back(stoi(k));
                    k = "";
                } else {
                    k += s[j];
                }
            }
            a.push_back(tmp);
        }
    }
    sort(a.begin(), a.end(), cmp);
    for (auto ai : a) {
        for (auto aij : ai) {
            if (check[aij] == 0) {
                check[aij] = 1;
                answer.push_back(aij);
            }
        }
    }
    return answer;
}